src/main/java/com/shootoff/camera/cameratypes/PS3EyeCamera.java

```diff
@@ -0,0 +1,621 @@
+			autoGain.setText("Off");
+			gain.setDisable(false);
+			exposure.setDisable(false);
+
+		} else {
+			isAutoGainSet = true;
+			autoGain.setText("On");
+			gain.setDisable(true);
+			exposure.setDisable(true);
+		}
+		autoGain.setSelected(isAutoGainSet);
+
+		GridPane.setConstraints(autoGain, 1, 4);
+		grid.getChildren().add(autoGain);
+
+		autoGain.selectedProperty().addListener(new ChangeListener<Boolean>() {
+			public void changed(ObservableValue<? extends Boolean> ov, Boolean old_val, Boolean new_val) {
+				if (new_val) {
+					autoGain.setText("On");
+					gain.setValue(eyecamLib.ps3eye_get_parameter(ps3ID, eyecam.ps3eye_parameter.PS3EYE_GAIN));
+					gain.setDisable(true);
+					exposure.setDisable(true);
+					eyecamLib.ps3eye_set_parameter(ps3ID, eyecam.ps3eye_parameter.PS3EYE_AUTO_GAIN, 1);
+				} else {
+					eyecamLib.ps3eye_set_parameter(ps3ID, eyecam.ps3eye_parameter.PS3EYE_AUTO_GAIN, 0);
+					autoGain.setText("Off");
+					gain.setValue(eyecamLib.ps3eye_get_parameter(ps3ID, eyecam.ps3eye_parameter.PS3EYE_GAIN));
+					gain.setDisable(false);
+					exposure.setDisable(false);
+				}
+			}
+		});
+
+		stage.show();
+
+		stage.setOnCloseRequest((e) -> {
+			configIsOpen = false;
+		});
+
+	}// end launchcamerasettings
+
+	public boolean setState(CameraState cameraState) {
+		this.cameraState = cameraState;
+		switch (cameraState) {
+		case DETECTING:
+			break;
+		case CALIBRATING:
+			break;
+		case CLOSED:
+			close();
+			break;
+		case NORMAL:
+		default:
+			break;
+
+		}
+
+		this.cameraState = cameraState;
+
+		return true;
+	}
+
+	public CameraState getState() {
+
+		return cameraState;
+	}
+
+	public void setCameraEventListener(CameraEventListener cameraEventListener) {
+		this.cameraEventListener = Optional.ofNullable(cameraEventListener);
+	}
+
+	public long getCurrentFrameTimestamp() {
+		return currentFrameTimestamp;
+	}
+
+	public String getName() {
+		return "PS3Eye";
+	}
+
+	private native static void initialize();
+
+	private static boolean cameraAvailableNative() {
+		return initialized;
+	}
+
+	private int getViewWidth() {
+		return 640;
+	}
+
+	private int getViewHeight() {
+		return 480;
+	}
+
+	public static void closeMe() {
+		if (configIsOpen) {
+			configIsOpen = false;
+			stage.close();
+		}
+		if (closed)
+			return;
+
+		eyecamLib.ps3eye_close(ps3ID);
+		eyecamLib.ps3eye_uninit();
+		logger.debug("ps3eye closed and uninit called");
+		ps3ID = null;
+		closed = true;
+		opened = false;
+	}
+
+	public synchronized void close() {
+
+		closeMe();
+		if (closed) {
+			return;
+		} else {
+			eyecamLib.ps3eye_close(ps3ID);
+			eyecamLib.ps3eye_uninit();
+			logger.debug("ps3eye closed and uninit called");
+			ps3ID = null;
+		}
+	}
+
+	public boolean isOpen() {
+		if (ps3ID != null) {
+			opened = true;
+			return true;
+		} else {
+			opened = false;
+			return false;
+		}
+	}
+
+	@Override
+	public double getFPS() {
+		return ps3FPS;
+	}
+
+	protected void setFPS(double newFPS) {
```

Comment:
I extended calculatedFPSCamera and took out a lot of the functions that were duplicated.  I started from OptiTrackCamera so that is why it was not extended.